falcon/inspect.py

```diff
@@ -767,11 +801,20 @@ def _get_source_info(obj, default='[unknown file]'):
         # responders coming from cythonized modules will
         # appear as built-in functions, and raise a
         # TypeError when trying to locate the source file.
-        source_info = default
+        return default
     return source_info
 
 
-def _get_source_info_and_name(obj):
+def _get_source_info_and_name(
+    obj: ModuleType
+    | Type[Any]
+    | MethodType
+    | FunctionType
+    | TracebackType
+    | FrameType
+    | CodeType
+    | Callable[..., Any]
+) -> Tuple[Optional[str], str]:
```

Comment:
```suggestion
) -> Tuple[str, str]:
```

Reply:
I cannot since `source_info = _get_source_info(obj, None)` can result in None being returned.